addons/udes_stock/models/stock_picking.py

```diff
@@ -179,6 +186,14 @@ def get_next_picking_name(self, vals, picking_type=None):
                 return get_next_name(self, "stock.picking", sequence=ir_sequence)
         return super().get_next_picking_name(vals, picking_type=picking_type)
 
+    @api.depends('move_lines.product_id', 'move_lines.product_uom_qty', 'move_lines.product_id.weight')
+    def _compute_total_weight(self):
+        for picking in self:
+            total = 0.0
+            for move in picking.move_lines:
+                total += move.product_uom_qty * move.product_id.weight
+            picking.u_total_weight = total
```

Comment:
move_weights = [move.product_uom_qty*move.product_id.weight for move in picking.move_lines]
picking.u_total_weight = sum(move_weights)